.github/workflows/tests.yaml

```diff
@@ -55,8 +55,34 @@ jobs:
           poetry-version: "1.3.2"
       - name: Install tox
         run: pip install tox
-      - name: run tox test
-        run: tox -e test
+#      - name: run tox test
+#        run: tox -e test
+  e2e-test:
+    name: End to End testing
+    needs: [ lint, unit-test ]
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        backend-name:
+          - "stubbed"
+          - "qx_emulator"
```

Comment:
Wondering if this would indeed work ?

In qiskit-quantuminspire it is 

```
        backend-name:
          - "Stubbed"
          - "QX emulator"
```

Reply:
Yes this will not work out-the-box, but I used snake case here to adhere to the pennylane name formatting.  

After rebasing, I included the registration of these custom devices (see `__init__.py`) this should make it possible to use our custom pennylane devices.

Reply:
This way user should also be protected from using QIBackend from `qiskit_quantuminspire.qi_backend` directly.

Reply:
After rebasing I refactored the way we initialise the device.